apps/src/p5lab/spritelab/coreLibrary.js

```diff
@@ -247,11 +249,13 @@ function atTimeEvent(inputEvent, p5Inst) {
     // the time matches the event argument)
     if (worldTime === inputEvent.args.n && previousTime !== inputEvent.args.n) {
       // Call callback with no extra args
+      eventLog.push(`atTime: ${inputEvent.args.n}`);
```

Comment:
how will curriculum interact with these events?

asking because i'm wondering if the events in the log should be a different data structure to make that interaction easier or less error-prone than a string comparison

Reply:
I wanted to avoid objects for two reasons:
1. make the code easier to write for curriculum writers who are somewhat less familiar with object manipulation
2. keep the data passing back and forth across the interpreter boundary as simple as possible.

The validation code will almost always look something like this, which seems pretty reasonable
```
var spriteIds = getSpriteIdsInUse();
var eventLog = getEventLog();
for (var i = 0; i < spriteIds.length; i++) {
  var spriteId = spriteIds[i];
  if (eventLog.includes("whenClick: " + i)) {
    // clicked!
    ...
  }
}
```

Reply:
okay, cool -- that does seem reasonable. i mostly wanted to make sure we thought about alternatives 😄